libindy/src/services/anoncreds/verifier.rs

```diff
@@ -127,4 +139,442 @@ impl Verifier {
 
         Ok(predicates_for_credential)
     }
+
+    pub fn verify_requested_restrictions(&self,
+                                         prover_service: &Prover,
+                                         proof_req: &ProofRequest,
+                                         full_proof: &Proof,
+                                         schemas: &HashMap<String, SchemaV1>,
+                                         cred_defs: &HashMap<String, CredentialDefinitionV1>) -> IndyResult<bool> {
+        let proof_attr_indexes = full_proof.requested_proof.unrevealed_attrs
```

Comment:
This isn't the same comparison. This functionality collects all identifier indexes for a specific requested attribute or predicate so that the filters can be verified.